src/actions/product-db-actions.ts

```diff
@@ -0,0 +1,91 @@
+import {Product} from '../db-schemas/product-schema';
+import {ProductDocument} from '../db-schemas/product-document';
+import {logger} from '../logger/logger';
+
+export const retrieveAllProducts = async () => {
+    const products = await Product.find();
+    return products.map((product) => product.toClient());
+};
+
+export const retrieveSpecificProduct = async (id: string) => {
+    const product = await Product.findById(id);
+    if (!product)
+        return null;
+    return product.toClient();
```

Comment:
```suggestion
    return product?.toClient();
```
instead of null checking

Reply:
same next methods

Reply:
for some reason I get an error:
  return product?.toClient();
                   ^
SyntaxError: Unexpected token '.'

So I did:
return product? product.toClient() : null;

Reply:
just update the TS version and restart WebStorm